### ISSUE TITLE
BetaDistribution policy for bounded continuous action spaces to avoid Gaussian clipping bias and improve training stability

### ISSUE THREAD

**Issue by lukaskiss222** [DLR-RM/stable-baselines3#2142]
### 🚀 Feature

Add a option of  `BetaDistribution` policy for bounded continuous action spaces to avoid Gaussian clipping bias and improve training stability.

### Motivation

[ Petrazzini & Antonelo (2021) ](https://arxiv.org/pdf/2111.02202) demonstrated that replacing Gaussian with a Beta distribution (compact support) in PPO yields significantly faster convergence, higher final rewards, and a 63 % increase in success on CarRacing-v0.

### Pitch

New `BetaDistribution` class in `stable_baselines3.common.distributions`, mirroring `DiagGaussianDistribution` with methods and optionally to use it on any `on policy algorithm`.

### Alternatives

- **Tanh-squashed Gaussian (SquashedDiagGaussianDistribution):** still biases density near ±1 and can have vanishing gradients at the tails .
- **Truncated normal**: more complex to implement and less stable under backpropagation .
- **State-Dependent Noise (gSDE)**: adapts variance but still uses Gaussian support, so does not fully eliminate bias at hard bounds .

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [x] If I'm requesting a new feature, I have proposed alternatives

**Comment by araffin** [comment]
Hello,
sorry for the delay.

> demonstrated that replacing Gaussian with a Beta distribution

Could you open a draft PR and run some benchmark? (checking that we can reproduce some results of the paper)
For some reference, you can have a look at what I did recently for checking the effect of n-step replay buffer (https://github.com/DLR-RM/stable-baselines3/pull/2144#issuecomment-2976012337), or have a look at the PR template of SB3-contrib, for instance https://github.com/Stable-Baselines-Team/stable-baselines3-contrib/pull/243

Please use the RL Zoo for that.

**Comment by araffin** [comment]
Some update, I used zed + Claude to create a proof of concept in https://github.com/DLR-RM/stable-baselines3/pull/2228
It does seem to learn something on the `Pendulum-v1` task but it is not extensively tested otherwise.

**Comment by lukaskiss222** [comment]
Hi,
sorry for delay @araffin .

I proposed it here, but never finished it. 

Can I continue on your proof of concept in #2228  and run  the required benchmarks? (CarRacing, LunarLander).

**Comment by lukaskiss222** [comment]
Hi @araffin,
I tried to replicate it as close as possible, but I used CarRacing-v3 instead of v0. Unfortunately, I was not able to reproduce the same scores, but at least I was able to reproduce better overall performance with Beta distribution.
One small change:
- For deterministic I used mean instead of mode for Beta distribution, which follows the paper implementation.

<img width="2700" height="1560" alt="Image" src="https://github.com/user-attachments/assets/cfc7df02-d816-4588-acd6-dd80ee35cda3" />

**Plot deterministic evaluation comparision:**

<img width="627" height="462" alt="Image" src="https://github.com/user-attachments/assets/76bb5b39-1c43-4e89-a594-960ca2802c9e" />

<img width="1952" height="502" alt="Image" src="https://github.com/user-attachments/assets/4099842b-2cfc-4e58-be52-90180e456bd4" />

<img width="1178" height="805" alt="Image" src="https://github.com/user-attachments/assets/655302d3-eb30-4cda-a1aa-9ce35bb22db1" />

<img width="627" height="473" alt="Image" src="https://github.com/user-attachments/assets/a8573452-1718-49d1-bb5e-eb20e537fd6f" />

<img width="984" height="790" alt="Image" src="https://github.com/user-attachments/assets/781943ba-93f8-4168-853e-68b7cfaabacd" />


**Gaussian**
```yaml
CarRacing-v3:
  env_wrapper:
    - beta_research.wrappers.CarRacingActionWrapper
    - beta_research.wrappers.ResizeGrayObservationWrapper
  env_kwargs:
    continuous: true
    domain_randomize: false
    lap_complete_percent: 1.0
  # 4 stacked 84x84 grayscale frames -> (4, 84, 84) input to NatureCNN.
  frame_stack: 4

  # Paper does not apply VecNormalize.
  normalize: False

  # PPO hyperparameters from Table II (CarRacing column).
  n_envs: 8
  n_timesteps: !!float 5e6
  policy: 'CnnPolicy'

  n_steps: 500
  batch_size: 64
  n_epochs: 10

  gamma: 0.99
  gae_lambda: 0.95

  # Adam lr = 2.5e-4 * alpha, alpha linearly annealed from 1 to 0.
  learning_rate: lin_2.5e-4

  clip_range: 0.1
  ent_coef: 0.01
  vf_coef: 0.5
  max_grad_norm: 0.5

  # Paper uses a plain Gaussian policy (no SDE).
  use_sde: False

  # Shared encoder: default NatureCNN (3 conv + linear -> 512) matches paper.
  # Actor head: 2 FC layers of 512.
  # Critic head: 1 FC layer of 512.
  policy_kwargs: "dict(net_arch=dict(pi=[512, 512], vf=[512]),
                       activation_fn=nn.Tanh)"

```
**Beta**
```yaml

CarRacing-v3:
  env_wrapper:
    - beta_research.wrappers.CarRacingActionWrapper
    - beta_research.wrappers.ResizeGrayObservationWrapper
  env_kwargs:
    continuous: true
    domain_randomize: false
    lap_complete_percent: 1.0

  frame_stack: 4
  normalize: False

  n_envs: 8
  n_timesteps: !!float 5e6
  policy: 'CnnPolicy'

  n_steps: 500
  batch_size: 64
  n_epochs: 10

  gamma: 0.99
  gae_lambda: 0.95

  learning_rate: lin_2.5e-4

  clip_range: 0.1
  ent_coef: 0.01
  vf_coef: 0.5
  max_grad_norm: 0.5

  use_sde: False

  # Same arch as Gaussian, with use_beta=True to switch the action distribution.
  policy_kwargs: "dict(net_arch=dict(pi=[512, 512], vf=[512]),
                       activation_fn=nn.Tanh,
                       use_beta=True)"

```
